src/lib/peer_trust/peer_trust.ml

```diff
@@ -46,15 +47,216 @@ end) (Now : sig
   val now : unit -> Time.t
 end)
 (Action : Action_intf)
-(Db : Key_value_database.S with type key := Peer.t and type value := Record.t) :
-  S with type peer := Peer.t and type action := Action.t = struct
-  type t = unit
+(Db : Key_value_database.S with type key := Peer.t and type value := Record.t) =
```

Comment:
The `Make` functor here takes `Peer` argument, which is anything hashable/sexpable.

I wonder if that kind of generality is needed / desirable. For the Merkle ledger tests, we made some effort to use "real" modules, after a period of using mock modules. Could we use `Network_peer.Peer`. 

Similarly, could we have a (perhaps-incomplete) version of `Action`, and use it, rather than a mock version?

For the db argument to the functor, that makes more sense, since the real db is slower, and isn't the critical bit being tested.

Reply:
I disagree. The generality isn't *needed*, but it's good for separation of concerns and we get something like parametricity from the functoring. From the signature you can tell this module doesn't do anything but hash and sexp-ize the peers. Actually, looking at it again we don't even hash them, so I can remove the `Hashable.S` include. Same thing with the actions.

I can easily imagine us adding persistent unique identifiers for peers and wanting to track trust based on those rather than IPs. This code shouldn't have to change for that.

I don't follow the argument for `Action`. Are you saying we import the real one and use it in the tests? That would cause a dependency cycle if it's defined in `coda_base`, which is the planned design.

Reply:
I think it's good here, but be careful -- parametricity via functor arguments can cause the nightmare of `coda_main.ml` if left unchecked. Parametricity via foralls is great though (`'a` basically, you can also do explicit foralls with `type a. a -> a` for example).

Why should `Action` be defined in `coda_base`?

Reply:
Ah. A good way to mitigate this (and what you should do I think) is keep all the functor arguments, and partially apply them to the concrete deps you have in scope here and export that. Minimizes pain on the functor-apply-side, but still keeps the nice parametricity.